### DISCUSSION TITLE
how to get JWT token for kubernetes auth method?

### DISCUSSION THREAD

**Discussion by vishalmehrahere** [rajanadar/VaultSharp#211] [Q&A] [answered]
how to get JWT token for kubernetes auth method in c#?
can u give me an example for getting this token?

**Comment by rajanadar** [comment, marked as answer]
@vishalmehra2214  The Kubernetes Service Account JWT token (for your pod) is outside the purview of VaultSharp.
Still, here are a couple of pointers.

Using CLI: Typically, you can get it as follows in 2 steps:

1. Get the secret name for your service account
```` kubectl get serviceaccounts <account-name> -o yaml ````

2. For the secret name, get the actual JWT token
````kubectl get secret <secret-name-from-above-output> -o yaml ````

And if you want it via C#, assuming your code runs in the pod, use a Kubernetes C# client like https://github.com/kubernetes-client/csharp to load the config file and get hold of the JWT token for your service account.

As I said, it is outside the purview of VaultSharp itself.

**Reply by vishalmehrahere** [reply]
Thank got my answer
